src/mediapipe_internal/pythonnoderesource.cpp

```diff
@@ -37,6 +37,32 @@ PythonNodeResource::PythonNodeResource() {
     this->nodeResourceObject = nullptr;
 }
 
+Status PythonNodeResource::finalize() {
```

Comment:
I don't think we need to return status. This could be void as we will not do anything with that status. Destruction will proceed regardless of finalize() operation success or failure. We would just log if any error is encountered.